libraries/node-core-library/src/FileSystem.ts

```diff
@@ -0,0 +1,432 @@
+ * @public
+ */
+export enum NewlineConversion {
+  CrLf,
+  Lf,
+  None
+}
+
+/**
+ * The options for FileSystem.readFolder()
+ * @public
+ */
+export interface IReadFolderOptions {
+  /**
+   * If true, returns the absolute paths of the files in the folder.
+   * Defaults to `false`.
+   */
+  absolutePath?: boolean;
+}
+
+/**
+ * The options for FileSystem.writeFile()
+ * @public
+ */
+export interface IWriteFileOptions {
+  /**
+   * If true, will ensure the folder is created before writing the file.
+   * Defaults to `false`.
+   */
+  ensureFolder?: boolean;
+
+  /**
+   * If specified, will normalize line endings to the specified style of newline.
+   * Defaults to `NewlineConversion.None`.
+   */
+  convertLineEndings?: NewlineConversion;
+
+  /**
+   * If specified, will change the encoding of the file that will be written.
+   * Defaults to `"utf8"`.
+   */
+  encoding?: Encoding;
+}
+
+/**
+ * The options for FileSystem.readFile()
+ * @public
+ */
+export interface IReadFileOptions {
+  /**
+   * If specified, will change the encoding of the file that will be written.
+   * Defaults to `"utf8"`.
+   */
+  encoding?: Encoding;
+
+  /**
+   * If specified, will normalize line endings to the specified style of newline.
+   * Defaults to `NewlineConversion.None`.
+   */
+  convertLineEndings?: NewlineConversion;
+}
+
+/**
+ * The options for FileSystem.move()
+ * @public
+ */
+export interface IMoveOptions {
+  /**
+   * If true, will overwrite the file if it already exists. Defaults to false.
+   */
+  overwrite?: boolean;
+
+  /**
+   * If true, will ensure the folder is created before writing the file.
+   * Defaults to `false`.
+   */
+  ensureFolder?: boolean;
+}
+
+/**
+ * The options for FileSystem.deleteFile()
+ * @public
+*/
+export interface IDeleteFileOptions {
+  /**
+   * If true, will throw an exception if the file did not exist before `deleteFile()` was called.
+   * Defaults to `false`.
+   */
+  throwIfNotExists?: boolean;
+}
+
+/**
+ * A standardized file system API. Mostly contains wrappers around `fs` and `fs-extra`.
+ * Also contains useful options to simplify and normalize file system operations.
+ * @public
+ */
+export class FileSystem {
+
+  // ===============
+  // COMMON OPERATIONS
+  // ===============
+
+  /**
+   * Returns true if the path exists on disk. It can be a file, folder, or link.
+   * Behind the scenes it uses `fs.existsSync()`.
+   * @param path - The absolute or relative path to the file, folder, or link.
+   */
+  public static exists(path: string): boolean {
+    return fsx.existsSync(path);
+  }
+
+  /**
+   * Gets the statistics for a particular file or folder.
+   * If the path is a link, this function follows the link and returns statistics about the link target.
+   * Behind the scenes it uses `fs.statSync()`.
+   * @param path - The absolute or relative path to the file, folder, or link.
+   */
+  public static getStatistics(path: string): fs.Stats {
+    return fsx.statSync(path);
```

Comment:
>statSync [](start = 15, length = 8)

This throws an exception if the object doesn't exist.

I found it useful at one point to have a tryGetStatistics() that returns undefined if the object doesn't exist.

Reply:
That's OK for now. We can add that later.

---
In reply to: [198664902](https://github.com/Microsoft/web-build-tools/pull/727#discussion_r198664902) [](ancestors = 198664902)